app/apollo-provider.tsx

```diff
@@ -0,0 +1,47 @@
+'use client'
+
+import { HttpLink, ApolloLink } from '@apollo/client'
+import {
+  NextSSRApolloClient,
+  ApolloNextAppProvider,
+  NextSSRInMemoryCache,
+  SSRMultipartLink
+} from '@apollo/experimental-nextjs-app-support/ssr'
+
+function apiEndpoint(): { uri: string; headers?: Record<string, string> } {
+  if (process.env.NODE_ENV === 'development') {
+    return { uri: 'http://127.0.0.1:54321/graphql/v1' }
+  } else {
+    return {
+      uri: process.env.GRAPHQL_ENDPOINT as string,
+      headers: {
+        apiKey: process.env.API_KEY as string
+      }
+    }
+  }
```

Comment:
```suggestion
   return {
      uri: process.env.GRAPHQL_ENDPOINT as string,
      headers: {
        apiKey: process.env.API_KEY as string
      }
    }
```

I'd say not to manage `NODE_ENV` could be better, because headers property can be undefined regardless of environment, and we didn't strictly check if the env values are set as string.